test/e2e/metamask-ui.spec.js

```diff
@@ -1510,17 +1516,11 @@ describe('MetaMask', function () {
     });
 
     it('finds the transaction in the transactions list', async function () {
-      await driver.wait(async () => {
-        const confirmedTxes = await driver.findElements(
-          '.transaction-list__completed-transactions .transaction-list-item',
-        );
-        return confirmedTxes.length === 5;
```

Comment:
Getting rid of these checks might end up bypassing the intent here, if the previous transaction happens to have the same text as the one this was supposed to wait for. Then the test would immediately find the wrong element and continue on.

It might be safer to restore these.

Reply:
I struggled with this one, I left it in a few cases where it was very hard to tell if the amount of ETH being sent was different. These tests are already not idempotent, relying on the order of operations to be successful. Skipping a test for expediency results in systematic failure of the entire spec after the last test. I'll restore these but I strongly feel like this whole file needs to be broken out into chunks.

Reply:
Maybe idempotency isn't a necessity in E2E tests, 🤔 it just feels like there is more than one thing being tested out here

Reply:
Oh for sure, these tests are a mess. They'll be broken out soon though hopefully!

Reply:
Done in latest